tests/PHPStan/Parser/TypeParserTest.php

```diff
@@ -276,6 +277,66 @@ public function provideParseData(): array
 				]),
 				Lexer::TOKEN_INTERSECTION,
 			],
+			[
+				'string~int',
+				new SubtractionTypeNode(
+					new IdentifierTypeNode('string'),
+					new IdentifierTypeNode('int')
+				),
+			],
+			[
+				'string ~ int',
+				new SubtractionTypeNode(
+					new IdentifierTypeNode('string'),
+					new IdentifierTypeNode('int')
+				),
+			],
+			[
+				'(string ~ int)',
+				new SubtractionTypeNode(
+					new IdentifierTypeNode('string'),
+					new IdentifierTypeNode('int')
+				),
+			],
+			[
+				'(' . PHP_EOL .
+				'  string' . PHP_EOL .
+				'  ~' . PHP_EOL .
+				'  int' . PHP_EOL .
+				')',
+				new SubtractionTypeNode(
+					new IdentifierTypeNode('string'),
+					new IdentifierTypeNode('int')
+				),
+			],
+			[
+				'string~int~float',
+				new SubtractionTypeNode(
+					new IdentifierTypeNode('string'),
+					new IdentifierTypeNode('int')
+				),
+				Lexer::TOKEN_SUBTRACTION,
+			],
+			[
+				'(string&int)~float',
```

Comment:
`~` should have higher precedence than `&` and `|`, therefore these:

```
A&B~C
A~B&C
A|B~C
A~B|C
```

should be tested